reasoner/resolution/framework/Request.java

```diff
@@ -133,7 +137,7 @@ public Path(Actor<? extends Resolver<?>> sender) {
             this(list(sender));
         }
 
-        private Path(List<Actor<? extends Resolver<?>>> path) {
+        public Path(List<Actor<? extends Resolver<?>>> path) {
             assert !path.isEmpty() : "Path cannot be empty";
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.